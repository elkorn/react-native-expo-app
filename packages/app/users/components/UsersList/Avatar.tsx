import React, { useState } from "react";
import { Image, StyleSheet, View, ActivityIndicator } from "react-native";
import { LoadingState } from "../../../shared/types";

export default function Avatar({ src }: AvatarProps) {
  const [loading, setLoading] = useState(LoadingState.Idle);

  return (
    <View style={styles.container}>
      <Image
        style={styles.image}
        onLoadStart={() => setLoading(LoadingState.Loading)}
        onLoad={() => setLoading(LoadingState.Success)}
        source={{
          uri: src,
        }}
      />
      <ActivityIndicator
        hidesWhenStopped
        animating={loading === LoadingState.Loading}
      />
    </View>
  );
}

interface AvatarProps {
  src: string;
}

const shape = {
  borderRadius: 100,
  width: 100,
  height: 100,
};

const styles = StyleSheet.create({
  container: {
    position: "relative",
    justifyContent: "center",
    alignItems: "center",
    marginRight: 12,
    ...shape,
  },
  image: {
    ...shape,
    position: "absolute",
    top: 0,
    left: 0,
    backgroundColor: "#eee",
  },
  error: {},
});
