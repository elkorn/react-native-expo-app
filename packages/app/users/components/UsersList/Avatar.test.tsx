import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import Avatar from "./Avatar";

describe("components", () => {
  describe("UsersList", () => {
    describe("Avatar", () => {
      it("should match snapshot", () => {
        const props = {
          src: " https://i.pravatar.cc/150?img=1",
        };

        const tree = renderer.create(<Avatar {...props} />);

        expect(tree).toMatchSnapshot();
      });
    });
  });
});
