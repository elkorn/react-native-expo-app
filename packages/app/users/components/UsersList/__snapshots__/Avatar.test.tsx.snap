// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components UsersList Avatar should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 100,
      "height": 100,
      "justifyContent": "center",
      "marginRight": 12,
      "position": "relative",
      "width": 100,
    }
  }
>
  <Image
    onLoad={[Function]}
    onLoadStart={[Function]}
    source={
      Object {
        "uri": " https://i.pravatar.cc/150?img=1",
      }
    }
    style={
      Object {
        "backgroundColor": "#eee",
        "borderRadius": 100,
        "height": 100,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": 100,
      }
    }
  />
  <ActivityIndicator
    animating={false}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
